std/math.zig

```diff
@@ -898,3 +898,58 @@ test "math.mulWide" {
     testing.expect(mulWide(i8, 5, -5) == -25);
     testing.expect(mulWide(u8, 100, 100) == 10000);
 }
+
+pub fn lessThan(a: var, b: var) bool {
+    const A = @typeOf(a);
+    const B = @typeOf(b);
+    if (@typeId(@typeOf(a)) != .Int or @typeId(@typeOf(b)) != .Int) @compileError("only integers supported");
+    if (A.is_signed == B.is_signed) return a < b;
+    if (A.is_signed) {
+        return @IntType(A.is_signed, A.bit_count + 1)(a) < @bitCast(@IntType(true, B.bit_count + 1), @IntType(B.is_signed, B.bit_count + 1)(b));
```

Comment:
Instead of this, I'd write:
```zig
if (a < 0 or maxInt(A) < b) return true;
const T = @IntType(false, A.bit_count);
return @truncate(T, a) < @truncate(T, b)
```